### ISSUE TITLE
CoreContentController action(s?) never called

### ISSUE THREAD

**Issue by lsoendergaard** [FluidTYPO3/fluidcontent_core#197]
After updating the following extensions to the specified versions:

- flux **7.2.3**
- fluidcontent **4.3.2**
- fluidcontent_core **1.2.0**
- fluidpages **3.3.1**

I'm using TYPO3 **6.2.15**

When clearing the cache on a page that has a shortcut fluidcontent_core element, I get the following error:

```
exec_SELECTquery
caller TYPO3\CMS\Core\Database\DatabaseConnection::exec_SELECTquery
ERROR You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near ') ORDER BY sorting ASC' at line 1
lastBuiltQuery SELECT * FROM tt_content WHERE uid IN () ORDER BY sorting ASC
```

The issue is, as you might be able to see, that there is no uid inside of the brackets of the `IN` clause, even though the content element definitely has a reference selected.

After digging a bit, I found out, that the error is caused by the `v:content.render` ViewHelper in the Shortcut.html Template, because the template variable `contentUids` is empty. I then simply tried to use DebuggerUtility inside the CoreContentController->shortcutAction and noticed that the action is never being called. I'm not exactly sure what versions I had before, but i think it was: flux 7.2.0, fluidcontent 4.2.4 and fluidcontent_core 1.1.3.

Other extensions: powermail, realurl.
I have no news extension installed (shouldn't be related to [this](https://github.com/FluidTYPO3/fluidcontent_core/pull/161)).
I have my own extension, that uses fluidcontent and fluidpages installed.

Additionally to the error above there is a nasty traceback that's probably not helpful:
```
debug_backtrace	require(typo3/sysext/cms/tslib/index_ts.php),index.php#28 // 
TYPO3\CMS\Frontend\Page\PageGenerator::renderContent#212 // 
TYPO3\CMS\Frontend\ContentObject\ContentObjectRenderer->cObjGet#214 // 
TYPO3\CMS\Frontend\ContentObject\ContentObjectRenderer->cObjGetSingle#697 // 
TYPO3\CMS\Frontend\ContentObject\UserContentObject->render#752 // 
TYPO3\CMS\Frontend\ContentObject\ContentObjectRenderer->callUserFunction#41 // 
call_user_func_array#6634 // 
TYPO3\CMS\Extbase\Core\Bootstrap->run# // 
TYPO3\CMS\Extbase\Core\Bootstrap->handleRequest#184 // 
TYPO3\CMS\Extbase\Mvc\Web\FrontendRequestHandler->handleRequest#195 // 
TYPO3\CMS\Extbase\Mvc\Dispatcher->dispatch#56 // 
TYPO3\CMS\Extbase\Mvc\Controller\ActionController->processRequest#69 // 
TYPO3\CMS\Extbase\Mvc\Controller\ActionController->callActionMethod#157 // 
call_user_func_array#286 //
FluidTYPO3\Flux\Controller\AbstractFluxController->renderAction# // 
FluidTYPO3\Flux\Controller\AbstractFluxController->performSubRendering#205 // 
TYPO3\CMS\Fluid\View\AbstractTemplateView->render#234 // 
FluidCache_Lfmtheme_Page_layout_Page_5ffc6371d6806a1ebc9bd14d9a83d16b0b2785e6->render#182 // 
TYPO3\CMS\Fluid\Core\ViewHelper\AbstractViewHelper->initializeArgumentsAndRender#152 // 
TYPO3\CMS\Fluid\Core\ViewHelper\AbstractViewHelper->callRenderMethod#230 // 
call_user_func_array#248 // 
TYPO3\CMS\Fluid\ViewHelpers\RenderViewHelper->render# // 
TYPO3\CMS\Fluid\View\AbstractTemplateView->renderSection#90 // 
FluidCache_Lfmtheme_Page_action_render_5639a6feedfdaac7d64b2831ef231788384e6ec5->section_4f9be057f0ea5d2ba72fd2c810e8d7b9aa98b469#222 // 
TYPO3\CMS\Fluid\Core\ViewHelper\AbstractViewHelper->initializeArgumentsAndRender#289 // 
TYPO3\CMS\Fluid\Core\ViewHelper\AbstractViewHelper->callRenderMethod#230 // 
call_user_func_array#248 // 
FluidTYPO3\Vhs\ViewHelpers\Content\RenderViewHelper->render# // 
FluidTYPO3\Vhs\ViewHelpers\Content\AbstractContentViewHelper->getContentRecords#45 //
FluidTYPO3\Vhs\ViewHelpers\Content\AbstractContentViewHelper->getRenderedRecords#125 // 
FluidTYPO3\Vhs\ViewHelpers\Content\AbstractContentViewHelper->renderRecord#210 // 
TYPO3\CMS\Frontend\ContentObject\ContentObjectRenderer->cObjGetSingle#242 // 
TYPO3\CMS\Frontend\ContentObject\RecordsContentObject->render#752 // 
TYPO3\CMS\Frontend\ContentObject\ContentObjectRenderer->cObjGetSingle#119 // 
TYPO3\CMS\Frontend\ContentObject\ContentObjectRenderer->cObjGetSingle#734 // 
TYPO3\CMS\Frontend\ContentObject\CaseContentObject->render#752 // 
TYPO3\CMS\Frontend\ContentObject\ContentObjectRenderer->cObjGetSingle#45 // 
TYPO3\CMS\Frontend\ContentObject\UserContentObject->render#752 // 
TYPO3\CMS\Frontend\ContentObject\ContentObjectRenderer->callUserFunction#41 // 
call_user_func_array#6634 // 
TYPO3\CMS\Extbase\Core\Bootstrap->run# // 
TYPO3\CMS\Extbase\Core\Bootstrap->handleRequest#184 // 
TYPO3\CMS\Extbase\Mvc\Web\FrontendRequestHandler->handleRequest#195 // 
TYPO3\CMS\Extbase\Mvc\Dispatcher->dispatch#56 // 
TYPO3\CMS\Extbase\Mvc\Controller\ActionController->processRequest#69 // 
TYPO3\CMS\Extbase\Mvc\Controller\ActionController->callActionMethod#157 // 
call_user_func_array#286 // 
FluidTYPO3\Flux\Controller\AbstractFluxController->renderAction# // 
FluidTYPO3\Flux\Controller\AbstractFluxController->performSubRendering#205 // 
TYPO3\CMS\Fluid\View\AbstractTemplateView->render#234 // 
FluidCache_FluidcontentCore_CoreContent_layout_NoHeader_a7a51fd73a841ef2fb491cafe73b2085a012d4f0->render#182 // 
TYPO3\CMS\Fluid\Core\ViewHelper\AbstractViewHelper->initializeArgumentsAndRender#247 // 
TYPO3\CMS\Fluid\Core\ViewHelper\AbstractViewHelper->callRenderMethod#230 // 
call_user_func_array#248 // 
FluidTYPO3\Flux\ViewHelpers\Form\DataViewHelper->render# // 
TYPO3\CMS\Fluid\Core\ViewHelper\AbstractViewHelper->renderChildren#89 // 
FluidCache_FluidcontentCore_CoreContent_layout_NoHeader_a7a51fd73a841ef2fb491cafe73b2085a012d4f0->{closure}#276 // 
TYPO3\CMS\Fluid\Core\ViewHelper\AbstractViewHelper->initializeArgumentsAndRender#235 // 
TYPO3\CMS\Fluid\Core\ViewHelper\AbstractViewHelper->callRenderMethod#230 // 
call_user_func_array#248 // 
FluidTYPO3\Vhs\ViewHelpers\TagViewHelper->render# // 
TYPO3\CMS\Fluid\Core\ViewHelper\AbstractViewHelper->renderChildren#43 // 
FluidCache_FluidcontentCore_CoreContent_layout_NoHeader_a7a51fd73a841ef2fb491cafe73b2085a012d4f0->{closure}#276 // 
TYPO3\CMS\Fluid\Core\ViewHelper\AbstractViewHelper->initializeArgumentsAndRender#223 // 
TYPO3\CMS\Fluid\Core\ViewHelper\AbstractViewHelper->callRenderMethod#230 // 
call_user_func_array#248 // 
TYPO3\CMS\Fluid\ViewHelpers\RenderViewHelper->render# // 
TYPO3\CMS\Fluid\View\AbstractTemplateView->renderSection#90 // 
FluidCache_FluidcontentCore_CoreContent_action_render_90d83a9f4d29cadcf29fb70b9b684114d6225ad0->section_62bce9422ff2d14f69ab80a154510232fc8a9afd#222 // 
TYPO3\CMS\Fluid\Core\ViewHelper\AbstractViewHelper->initializeArgumentsAndRender#135 // 
TYPO3\CMS\Fluid\Core\ViewHelper\AbstractViewHelper->callRenderMethod#230 // 
call_user_func_array#248 // 
FluidTYPO3\Vhs\ViewHelpers\Content\RenderViewHelper->render# // 
FluidTYPO3\Vhs\ViewHelpers\Content\AbstractContentViewHelper->getContentRecords#45 // 
FluidTYPO3\Vhs\ViewHelpers\Content\AbstractContentViewHelper->getContentRecordsFromPage#114 // 
TYPO3\CMS\Core\Database\DatabaseConnection->exec_SELECTgetRows#175 // 
TYPO3\CMS\Core\Database\DatabaseConnection->exec_SELECTquery#370 // 
TYPO3\CMS\Core\Database\DatabaseConnection->debug#305
```

**Comment by lsoendergaard** [comment]
@NamelessCoder could you reopen this? My issue seems to be something else.

After digging a bit, I found out that  it tries to call the default action `render` instead of `shortcut`. `pluginName` etc. is set correctly to `Shortcut`, and it seems to be in [`AbstractFluxController.php`](https://github.com/FluidTYPO3/flux/blob/development/Classes/Controller/AbstractFluxController.php#L223) where the decision is made to use render. `$shouldRelay` is `false` because `$isForeign` in `hasSubControllerActionOnForeignController` is `false` (`$this->extensionName` is `"FluidcontentCore"` and `$extensionName` is `"FluidTYPO3.FluidcontentCore"`).

**Comment by jmverges** [comment]
Hi there, 

I can confirm @lsoendergaard issue.  I'm going to reopen this issue but I think the fix should be done in Flux. This commit https://github.com/FluidTYPO3/flux/commit/eb2abb03e12fa5d367854d580fbeb33908520e9d makes the changes.

Here is the screenshot https://cdn.pbrd.co/images/1H4DEB0t.png

I don't why and when is necessary $isForeign variable, @NamelessCoder could you bring some light about this?

**Comment by jmverges** [comment]
maybe related with: #196

**Comment by lsoendergaard** [comment]
Reverting the change and relying on `"FluidcontentCore"` being different from `"FluidTYPO3.FluidcontentCore"` seems a bit odd. It looks more like "being foreign" is not a reliable enough condition to use here.

**Comment by jmverges** [comment]
@lsoendergaard I agree with you, lets wait @NamelessCoder brings some light here ;)

**Comment by mischka** [comment]
Is there a plan to get this fix to the 7.2.* branch for the 6.2 LTS?

**Comment by NamelessCoder** [comment]
> Is there a plan to get this fix to the flux 7.2.* branch for the 6.2 LTS?

See https://github.com/FluidTYPO3/flux/issues/1149#issuecomment-223269520 for info that applies to every repository.
